### PR TITLE
Adding first 6 methods for iteration 4, and broken-out modules for al…

### PR DESCRIPTION
**Pull request by aprildagonese**
Adds Iteration 4 methods for the following:
- team_info
- best_season
- worst_season
- average_win_percentage
- most_goals_scored
- fewest_goals_scored

Breaks out modules for Team, Game, and GameTeam, as well as their test files, but does not move code into them (except Team).

@allen4397 @teresa-m-knowles I thought I was going to have to put some of these methods into the Game and GameTeams modules, so I started to set up those files for this. But then I decided to just throw them all into the TeamStats module and talk to you guys about it later. So the files are in there, but mostly untouched for now.

Also, I had to use the win_percentages method off StatTracker instead of Team because I needed to feed it a team_id and didn't have a team instance available. So for now I'm still using those, but we can talk about how to consolidate. I still can't get win_percentages working, though! It's still returning 75% when I expect 80%. Did either of you end up figuring that out?

### CODE REVIEW

**Comment by teresa-m-knowles** [comment]
@aprildagonese Yes! So the win_percentage is now a team method and we fixed the math so now it works. It's on the pull request that Tim just did

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -56,30 +71,22 @@ def percentage_home_wins
...
-
   def average_goals_per_game
     total_goals = @games.inject(0) do |sum, game|
       sum + game.total_score
     end
     (total_goals.to_f/@games.length.to_f).round(2)
   end
 
-  def games_by_season
+  def all_games_by_season
```
Comment (teresa-m-knowles) [inline lib/stat_tracker.rb]: good change, this is more explicit

Comment (allen4397) [reply lib/stat_tracker.rb]: Agreed

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -3,12 +3,18 @@
...
 
 class StatTracker
+  include TeamStats, GameStats, GameTeamStats
+
   attr_accessor :teams,
                 :games
 
-  attr_reader :game_teams
+  attr_reader :game_teams,
+              :team_stats
```
Comment (allen4397) [inline lib/stat_tracker.rb]: Just noticed this team_stats attr. What instance variable is this connected to? Or what does this return when called?

File: lib/team_stats.rb [hunk 3]
```diff
@@ -0,0 +1,81 @@
+module TeamStats
+
+  def test_module
+    "THIS MODULE WORKS!"
+  end
+
```
Comment (allen4397) [inline lib/team_stats.rb]: This is all outstanding. Thanks for your work on this.

**Comment by allen4397** [comment]
How do we want to approach the merge conflicts? Maybe we can just look at it and it will seem straightforward, unless there is a weird change?

**Comment by allen4397** [comment]
This all looks good. I reviewed the changes and there shouldn't be anything that breaks anything. There may be one test that had a change in value, so I will look at the testing in case something changed but the code for all the methods looks great!

**Pull request merged**